### PR TITLE
Fix MasterDetailPage/NavigationPage leaks on iPad

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Fixes leaks preventing MasterDetailPage and its child pages from being garbage collected on iPad.

### Bugs Fixed ###

- [44166 – but on the iPad](https://bugzilla.xamarin.com/show_bug.cgi?id=44166)

### API Changes ###

None

### Behavioral Changes ###

None

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla44166.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ public _44166MDP()
 
 			Master = new _44166Master();
 			Detail = new _44166Detail();
-		}
+        }
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla44166.cs]: um...spaces?

Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla44166.cs]: Yeah, had to reinstall VS yesterday, and spaces is the default in VS. Thought I'd caught all of them, but I guess not.

Comment (StephaneDelcroix) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla44166.cs]: from now on, I'll refer to @samhouts as Style Cop 👮

File: Xamarin.Forms.Platform.iOS/Renderers/TabletMasterDetailRenderer.cs [hunk 2]
```diff
@@ -71,36 +68,50 @@ UIBarButtonItem PresentButton
...
+				if (_masterController != null)
+				{
+					_masterController.WillAppear -= MasterControllerWillAppear;
+					_masterController.WillDisappear -= MasterControllerWillDisappear;
+				}
+
+				ClearControllers();
+			}
+
+			base.Dispose(disposing);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.iOS/Renderers/TabletMasterDetailRenderer.cs]: this is reshuffling to reduce nesting, right? or is there any change in it ?

Comment (hartez) [inline Xamarin.Forms.Platform.iOS/Renderers/TabletMasterDetailRenderer.cs]: It's a fix to the `Dispose` method not marking the item as `_dispose = true;` when running from the finalizer, plus it adds the `ClearControllers()` call that was missing from the tablet version.

**Pull request merged**